components/post_view/channel_intro_message/pluggable_intro_buttons/pluggable_intro_buttons.tsx

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+
+import {Channel} from '@mattermost/types/channels';
+
+import {PluginComponent} from 'types/store/plugins';
+
+type Props = {
+    channel: Channel;
+    pluginButtons: PluginComponent[];
+}
+
+const PluggableIntroButtons = React.memo((props: Props) => {
+    const intl = useIntl();
+
+    const channelIsArchived = props.channel.delete_at !== 0;
+    if (channelIsArchived || props.pluginButtons.length === 0) {
+        return null;
+    }
+
+    const buttons = props.pluginButtons.map((buttonProps) => {
+        // text defaults to "Create a board" because the original version of this API had hardcoded text and was
+        // built specifically for Boards
+        const text = buttonProps.text ?? intl.formatMessage({id: 'intro_messages.createBoard', defaultMessage: 'Create a board'});
+
+        return (
+            <button
+                key={buttonProps.id}
+                className={'intro-links color--link channelIntroButton style--none'}
+                onClick={buttonProps.action}
```

Comment:
Should we consider providing channel context specifics to reduce the need to go select these manually? (like in `registerCallButtonAction`, the action there has a signature of `(currentChannel: Channel, myCurrentChannelMembership: ChannelMembership) => void)`

Reply:
Yeah, we probably should. I removed the comment about that from the registry method because we weren't actually passing them, but we probably need to to make this at all useful. I also remember looking at how the Boards code that uses this works, and it immediately reads Redux to get that information which is something we'd definitely like to reduce the need for